dlt/destinations/utils.py

```diff
@@ -23,6 +23,12 @@ def ensure_resource(data: Any) -> DltResource:
     """Wraps `data` in a DltResource if it's not a DltResource already."""
     if isinstance(data, DltResource):
         return data
+    # prevent accidentally wrapping sources with adapters
+    if isinstance(data, DltSource):
+        raise Exception(
+            "You are trying to use an adapter on a dlt source. You can only use adapters on pure"
```

Comment:
Is this function only used with adapters? The error message seems to be out of context for `ensure_resource` function. Also, `TypeError` maybe?

Reply:
Yes, it only is used in adapters, I checked that.